tests/rptest/scale_tests/basic_auth_test.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import json
+import requests
+import threading
+from ducktape.mark import matrix
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.admin import Admin
+from rptest.services.redpanda import SecurityConfig
+from rptest.tests.pandaproxy_test import PandaProxyEndpoints, Consumer
+from rptest.util import wait_until_result
+from typing import Optional, List, Dict, Union
+
+
+class User:
+    def __init__(self):
+        self.username = f'user_{TopicSpec()._random_topic_suffix()}'
+        self.password = f'secret_{self.username}'
+        self.algorithm = 'SCRAM-SHA-256'
+
+
+class GetTopics(threading.Thread):
+    def __init__(self, thid: int, user: User, handle):
+        threading.Thread.__init__(self)
+        self.thid = thid
+        self.user = user
+        self._get_topics = handle
+        self.result_raw = None
+
+    def run(self):
+        self.result_raw = self._get_topics(auth=(self.user.username,
+                                                 self.user.password))
+
+
+class BasicAuthScaleTest(PandaProxyEndpoints):
+    topics = [
+        TopicSpec(partition_count=1, replication_factor=1),
+    ]
+
+    def __init__(self, context):
+
+        security = SecurityConfig()
+        security.enable_sasl = True
+        security.endpoint_authn_method = 'sasl'
+        security.pp_authn_method = 'http_basic'
+
+        super(BasicAuthScaleTest, self).__init__(
+            context,
+            security=security,
+            pp_keep_alive=60000 * 5,  # Time in ms
+            pp_cache_max_size=100)
```

Comment:
Solving one concurrency problem led to another:
```
redpanda: /home/nyalia/redpanda/vbuild/debug/clang/v_deps_build/seastar-prefix/src/seastar/include/seastar/core/timer.hh:95: void seastar::timer<seastar::lowres_clock>::arm_state(seastar::timer::time_point, std::optional<duration>) [Clock = seastar::lowres_clock]: Assertion `!_armed' failed.
```

This tells me that the garbarge collection timer is `rearm'd` after it was already armed. This is likely happening because of eviction so perhaps there is a race to "who triggers garbarge collection first."